### PR TITLE
996: Extend to support for t2a cases

### PR DESCRIPTION
**Pull request by ddebella**

### CODE REVIEW

File: migrations/app/20171010160121_app_workload_points_t2a_column.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+exports.up = function (knex, promise) {
+  var sql = 'ALTER TABLE app.workload_points ADD isT2A BIT;'
```
Comment (andrew-js-wright) [inline migrations/app/20171010160121_app_workload_points_t2a_column.js]: in order to keep consistent with the rest of our fields this should be in snake case i.e. `is_t2a`.

File: test/helpers/data/staging-workload-helper.js [hunk 2]
```diff
@@ -111,24 +114,45 @@ module.exports.deleteAll = function () {
     })
 }
 
-function mapForInsert (record) {
+function mapForInsert (record, isT2A = false) {
   var row = {}
   for (let key in record) {
-    if (key === 'communityTiers') {
+    if (key === 'communityTiers' && !isT2A) {
```
Comment (andrew-js-wright) [inline test/helpers/data/staging-workload-helper.js]: I don't see how the mapping process is different for t2a cases?

File: test/integration/services/data/test-get-staging-workload.js [hunk 3]
```diff
@@ -13,6 +13,9 @@ describe('services/data/get-staging-workload', function () {
   before(function () {
     return workloadHelper.insertCaseSummaryReport(caseSummaryReport, insertedRecords)
     .then(function (inserts) {
+      return workloadHelper.insertCaseSummaryReport(caseSummaryReport, insertedRecords, true)
```
Comment (andrew-js-wright) [inline test/integration/services/data/test-get-staging-workload.js]: Can you use a variable here so it is clear this `true` relates to T2A?

File: test/helpers/data/staging-workload-helper.js (hunk 2 above)
Comment (ddebella) [reply test/helpers/data/staging-workload-helper.js]: the mapping is based on the key. t2a cases will contain case summary of t2aCommunityTiers, etc. The underline mapping is the same. We need this to ensure we only insert either t2a cases or non-t2a cases.

File: migrations/app/20171010160121_app_workload_points_t2a_column.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  return knex.schema
+        .raw('SET ARITHABORT ON')
+        .raw(sql)
+}
+
+exports.down = function (knex, Promise) {
+  var sql = 'ALTER TABLE app.workload_points DROP COLUMN isT2A;'
```
[the remark this answers is not in the document]
Comment (ddebella) [reply migrations/app/20171010160121_app_workload_points_t2a_column.js]: I think this file was created before deciding on the approach of showing the total cases in the workload table directly. I don't believe we need this any longer.

File: migrations/app/20171010160121_app_workload_points_t2a_column.js (hunk 1 above)
Comment (ddebella) [reply migrations/app/20171010160121_app_workload_points_t2a_column.js]: 👍

File: app/services/data/get-staging-workload.js [hunk 5]
```diff
@@ -11,6 +11,7 @@ const Promise = require('bluebird').Promise
 module.exports = function (range) {
   var omWorkloads = []
   return knex('wmt_extract').whereBetween('wmt_extract.id', range)
+    .leftJoin('t2a', 'wmt_extract.om_key', 't2a.om_key')
```
Comment (caoimheb) [inline app/services/data/get-staging-workload.js]: `om_key` does not uniquely define a workload owner. Think you also need to join on team here

File: migrations/app/20171011120955_app_workload_add_t2a_column.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+exports.up = function (knex, promise) {
+  var sql = 'ALTER TABLE app.workload ADD t2a_cases INT;'
+  return knex.schema
+        .raw('SET ARITHABORT ON')
+        .raw(sql)
+}
+
+exports.down = function (knex, Promise) {
+  var sql = 'ALTER TABLE app.workload_points DROP COLUMN t2a_cases;'
```
Comment (caoimheb) [inline migrations/app/20171011120955_app_workload_add_t2a_column.js]: Should be dropping from the `workload` table

File: package.json [hunk 7]
```diff
@@ -56,6 +56,6 @@
     "mssql": "3.3.0",
     "request": "^2.81.0",
     "request-promise": "^4.2.1",
-    "wmt-probation-rules": "ministryofjustice/wmt-probation-rules.git#v0.2.0"
+    "wmt-probation-rules": "ministryofjustice/wmt-probation-rules.git#v0.3.0"
```
Comment (caoimheb) [inline package.json]: This will need bumped to match your updated probation rules version

File: seed/data/dev/00800_workload_points.js [hunk 8]
```diff
@@ -40,7 +40,48 @@ exports.seed = function (knex, Promise) {
...
+          default_contracted_hours_po: 33,
+          default_contracted_hours_pso: 33,
+          weighting_o: 88,
+          weighting_w: 92,
+          weighting_u: 96,
+          weighting_arms_lic: 96,
+          weighting_arms_comm: 96,
+          paroms_enabled: 1,
+          parom: 119,
+          is_t2a: true
```
Comment (caoimheb) [inline seed/data/dev/00800_workload_points.js]: 👍

File: seed/data/dev/00800_workload_points.js [hunk 9]
```diff
@@ -40,7 +40,48 @@ exports.seed = function (knex, Promise) {
...
+          nominal_target_spo: 2172,
+          nominal_target_po: 2172,
+          default_contracted_hours_po: 33,
+          default_contracted_hours_pso: 33,
+          weighting_o: 88,
+          weighting_w: 92,
+          weighting_u: 96,
+          weighting_arms_lic: 96,
+          weighting_arms_comm: 96,
+          paroms_enabled: 1,
```
Comment (caoimheb) [inline seed/data/dev/00800_workload_points.js]: `paroms_enabled` can be removed. We don't insert or use it

File: seed/views/00502_ldu_capacity_breakdown_view.js [hunk 10]
```diff
@@ -22,8 +23,9 @@ exports.seed = function (knex, promise) {
       , SUM(tcbv.cms_adjustment_points) AS cms_adjustment_points
       , SUM(tcbv.gs_adjustment_points) AS gs_adjustment_points
       , SUM(tcbv.total_cases) AS total_cases
+      , SUM(tcbv.t2a_cases) AS t2a_cases
       , SUM(tcbv.contracted_hours) AS contracted_hours
-      , t.id as id
+      , t.id AS id
       , tcbv.grade_code
```
Comment (caoimheb) [inline seed/views/00502_ldu_capacity_breakdown_view.js]: 👍  🙂

**Comment by caoimheb** [comment]
Do we also not need to include T2A cases when calculating the workload points?

File: app/services/data/get-staging-workload.js (hunk 5 above)
Comment (andrew-js-wright) [reply app/services/data/get-staging-workload.js]: Examples of this should be in master now.

File: seed/data/dev/00800_workload_points.js (hunk 9 above)
Comment (caoimheb) [reply seed/data/dev/00800_workload_points.js]: Actually, looks like it can't be nullable. Nvm

**Pull request closed without merging**